src/init/init.ts

```diff
@@ -177,7 +177,7 @@ export async function runGenerator(generatorName: string, options: {[name: strin
   options = options || {};
   const templateName = options['templateName'] || generatorName;
 
-  const env: YeomanEnvironment = await options['env'] || createYeomanEnvironment();
+  const env: YeomanEnvironment = await options['env'] || await createYeomanEnvironment();
```

Comment:
Options['env'] is always a non-Promise value. The `await` was intended to apply to the whole or'ed expression. I'd suggest keeping just one `await` by doing: 

```
const env: YeomanEnvironment = await (options['env'] || createYeomanEnvironment());
```